# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/linux/amd-x2apic-cpufreq.patch
# Copyright (C) 2021 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- linux-5.14/drivers/acpi/processor_core.c	2021-08-30 00:04:50.000000000 +0200
+++ linux-5.14/drivers/acpi/processor_core.c	2021-09-26 18:42:00.234480106 +0200
@@ -54,7 +57,7 @@
 	if (!(apic->lapic_flags & ACPI_MADT_ENABLED))
 		return -ENODEV;
 
-	if (device_declaration && (apic->uid == acpi_id)) {
+	if (apic->uid == acpi_id) {
 		*apic_id = apic->local_apic_id;
 		return 0;
 	}
--- linux-5.14/drivers/acpi/acpi_processor.c	2021-08-30 00:04:50.000000000 +0200
+++ linux-5.14/drivers/acpi/acpi_processor.c	2021-09-26 18:44:44.540473555 +0200
@@ -277,7 +279,7 @@
 	pr->phys_id = acpi_get_phys_id(pr->handle, device_declaration,
 					pr->acpi_id);
 	if (invalid_phys_cpuid(pr->phys_id))
-		dev_dbg(&device->dev, "Failed to get CPU physical ID.\n");
+		dev_err(&device->dev, "Failed to get CPU physical ID.\n");
 
 	pr->id = acpi_map_cpuid(pr->phys_id, pr->acpi_id);
 	if (!cpu0_initialized && !acpi_has_cpu_in_madt()) {
